src/client_side.cc

```diff
@@ -1323,8 +1330,9 @@ ClientSocketContext::buildRangeHeader(HttpReply * rep)
         // web server responded with a valid, but unexpected range.
         // will (try-to) forward as-is.
         //TODO: we should cope with multirange request/responses
-        bool replyMatchRequest = rep->content_range != NULL ?
-                                 request->range->contains(rep->content_range->spec) :
+        // TODO: review, since rep->content_range is always nil here.
+        bool replyMatchRequest = contentRange != nullptr ?
```

Comment:
This can be better written as:
 bool replyMatchRequest = !contentRange || request->range->contains(contentRange->spec);

Reply:
@yadij, please note that I missed this change request in the withdrawn review. I see a few `nullptr` callers in v3.5, so perhaps this is not important, but if you do want them removed from this PR, please let Eduard know. It should be a simple change.